src/encoded/viewconfigs/views.py

```diff
@@ -0,0 +1,109 @@
+from pyramid.view import view_config
+from snovault.viewconfigs.searchview import SearchView
+from snovault.resource_views import collection_view_listing_db
+from snovault.viewconfigs.report import ReportView
+from encoded.viewconfigs.news import NewsView
+from encoded.viewconfigs.matrix import MatrixView
+from encoded.viewconfigs.auditview import AuditView
+from encoded.viewconfigs.summary import SummaryView 
+from encoded.helpers.helper import (View_Item, search_result_actions)
+from snovault import AbstractCollection
+
+
+def includeme(config):
+    config.add_route('search', '/search{slash:/?}')
+    config.add_route('report', '/report{slash:/?}')
+    config.add_route('matrix', '/matrix{slash:/?}')
+    config.add_route('news', '/news/')
+    config.add_route('audit', '/audit/')
+    config.add_route('summary', '/summary{slash:/?}')
+    config.scan(__name__)
+
+def _get_doc_types(context, request):
+    doc_types = []
+
+    ## converts type= query parameters to list of doc_types to search, "*" becomes super class Item
+    if (hasattr(context, 'type_info') and hasattr(context.type_info, 'name') and context.type_info.name):
+        doc_types = [context.type_info.name]
+    else:
+        doc_types = request.params.getall('type')
+    
+    if '*' in doc_types:
+        doc_types = ['Item']
+    
+    return doc_types
+
+def iter_search_results(context, request):
+    return search(context, request, return_generator=True)
+
+@view_config(context=AbstractCollection, permission='list', request_method='GET', name='listing')
+def collection_view_listing_es(context, request):
+    # Switch to change summary page loading options
+    if request.datastore != 'elasticsearch':
+        return collection_view_listing_db(context, request)
+    return search(context, request)
+
+@view_config(route_name='search', request_method='GET', permission='search')
+def search(context, request, search_type=None, return_generator=False):
+    default_doc_types = [
```

Comment:
default_doc_types was made an argument because many other objects call SearchView and they have different values for default_doc_types. So, adding it as a class variable in SearchView may not meet the needs of all objects and introduce encode-specific settings into snovault.  Does this make sense to you?

Reply:
Can you list the locations where this occurs?

Also, did you fix the default_data_types?

Reply:
Phil- Note clear.